### PR TITLE
[2.19.x] G-8428 Refactored Logout Redirect

### PR DESCRIPTION
**Pull request by brianfelix**
#### What does this PR do?
Refactored Logout Redirect to use the search UI config and apply to the search UI logout button.

#### Who is reviewing it? 
@mojogitoverhere 
@andrewzimmer 
@pklinef 

#### How should this be tested?
1.) Verify ddf behavior remains unchanged if the new config is not set:
Navigate to the Search UI
Open Chrome Dev tools or a similar function in your browser in order to see network requests
Use the signout button at the top right of the search UI to logout
Verify there is a network request with a name that starts with "main.js"

2.) Verify setting the config changes the signout button of the search UI to immediately redirect the user to the logout url instead of processing the logout action providers:
Navigate to the Admin UI
Click on the "System" tab
Click on the configuration titled "Catalog UI Search"
Verify the value for the field "Logout Url" is empty by default
Set the Logout Url to the single character "/"
Navigate to the search UI
Open Chrome Dev tools or a similar function in your browser in order to see network requests
Use the signout button at the top right of the search UI to logout
Verify there is NOT a network request with a name that starts with "main.js"
Verify you are redirected to https://localhost:8993/

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by brianfelix** [comment]
build now

File: platform/security/servlet/security-servlet-logout/src/main/java/org/codice/ddf/security/servlet/logout/LocalLogoutServlet.java [hunk 1]
```diff
@@ -50,14 +45,11 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response)
...
-        String.format(
-            "{ \"mustCloseBrowser\": %b, \"redirectUrl\": \"%s\" }",
-            mustCloseBrowser, getRedirectUrlMessage());
+    String message = String.format("{ \"mustCloseBrowser\": %b }", mustCloseBrowser);
 
-    response.setStatus(HttpServletResponse.SC_OK);
-    response.setContentType("application/json");
     try {
+      response.setStatus(HttpServletResponse.SC_OK);
+      response.setContentType("application/json");
```
Comment (Lambeaux) [inline platform/security/servlet/security-servlet-logout/src/main/java/org/codice/ddf/security/servlet/logout/LocalLogoutServlet.java:52]: ❓ These setters can't throw `IOException`s, correct?

**Comment by mazarag2** [comment]
Not sure If I'm missing something from my end but I pulled down the branch and built it with `profile:install development` when I go to  the search https://localhost:8993/search I'm seeing this behavior where it looks like its stuck on a infinite redirect loop 

https://user-images.githubusercontent.com/16808210/110996474-3cb80700-8339-11eb-873a-89ed5a8c7a34.mov

**Review by Lambeaux — APPROVED** [review]

File: platform/security/servlet/security-servlet-logout/src/main/java/org/codice/ddf/security/servlet/logout/LocalLogoutServlet.java (hunk 1 above)
Comment (brianfelix) [reply platform/security/servlet/security-servlet-logout/src/main/java/org/codice/ddf/security/servlet/logout/LocalLogoutServlet.java:52]: They cannot. The modifications to LocalLogoutServlet in this PR are reverting the module back to how it was before any work was done for the original logout redirect PR. I can keep the change fromt the old PR that takes these lines out of the try/catch

**Comment by brianfelix** [comment]
@mazarag2 I tried to reproduce the issue you saw but was unable to. Navigating to https://localhost:8993/search immediately redirects me to https://localhost:8993/catalog/search where I am able to interact with the search UI normally.

File: platform/security/servlet/security-servlet-logout/src/test/java/org/codice/ddf/security/servlet/logout/LogoutServletTest.java [hunk 2]
```diff
@@ -134,52 +128,7 @@ public void testLocalLogoutNotBasicOrPki() {
...
-    SecurityTokenHolder securityTokenHolder = mock(SecurityTokenHolder.class);
-    when(httpSession.getAttribute(SecurityConstants.SECURITY_TOKEN_KEY))
-        .thenReturn(securityTokenHolder);
-    try {
-      localLogoutServlet.doGet(request, response);
-    } catch (ServletException e) {
-      fail(e.getMessage());
-    }
-    verify(httpSession).invalidate();
-    verify(printWriter)
```
Comment (cassandrabailey293) [inline platform/security/servlet/security-servlet-logout/src/test/java/org/codice/ddf/security/servlet/logout/LogoutServletTest.java:159]: it looks like this test was actually validating a set "redirect url". is there any way we can add a test to validate the actual logout url set by the config? looks like most the tests here now are non happy path tests, or they just check the `mustCloseBrowser` bool but not the url.

**Review by cassandrabailey293 — COMMENTED** [review]
was the user able to configure the `redirectUrl` before? Now, the user can configure the logout url via the ui settings...are there any security implications with this? if not then it looks good to me.

**Review by pklinef — APPROVED** [review]

**Review by andrewzimmer — APPROVED** [review]

File: platform/security/servlet/security-servlet-logout/src/test/java/org/codice/ddf/security/servlet/logout/LogoutServletTest.java (hunk 2 above)
Comment (brianfelix) [reply platform/security/servlet/security-servlet-logout/src/test/java/org/codice/ddf/security/servlet/logout/LogoutServletTest.java:159]: This ticket was originally done last year by modifying the LogoutServlet module to set a redirect URL. That fix did not work in some downstream projects so this PR is reimplementing the logout redirect in a different way and undoing those previous changes in this module. I added the tests you mentioned when I worked on the previous fix for this but once this is merged, the "redirect URL" will no longer exist and the LogoutServlet module will not be involved in the signout button redirect

**Comment by brianfelix** [comment]
@cassandrabailey293 The redirect URL was made for a previous implementation of the work in this ticket. Now that it is being reimplemented, the redirect URL will not longer be used

**Comment by andrewzimmer** [comment]
Hero Successful 🎉 
- verified default ddf behavior remains the same
- verified when logout redirect set to "/" redirected to localhost:8993 and no request with main.js

**Comment by andrewzimmer** [comment]
build now

**Pull request merged**